### PR TITLE
[READY] Feature/create ahk run configs

### PR DESCRIPTION
**Pull request by takanuva15**
Hi mtrnord, this PR fixes the run config functionality so that we can run AHK scripts via a run config.  Basically you just create a new AutoHotkey SDK and then set that for the project, and then any scripts that are run will use that SDK.

I have also updated the plugin's compatible build range in the build script to include everything from 2019.3 to 2020.1.  

(Assuming everything looks good, I would like to update the release version to 0.2.0 and push that to JetBrains with this new stuff.)

### CODE REVIEW

File: src/main/java/de/nordgedanken/auto_hotkey/sdk/AhkSdkType.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	}
+
+	@Override
+	public @NotNull String suggestSdkName(@Nullable String currentSdkName, String sdkHome) {
+		return "AutoHotkey";
+	}
+
+	@Override
+	public final String getVersionString(String sdkHome) {
+		return "unknown version";
```
Comment (MTRNord) [inline src/main/java/de/nordgedanken/auto_hotkey/sdk/AhkSdkType.java:63]: Node for the future: Extract it somehow. Maybe using a small ahk script we ship and run.

**Comment by MTRNord** [comment]
grafik

I think this is introduced by your PR (I will double-check it). But it seems like the project doesn't have a file tree anymore if I use the open function in the welcome menu (not the import). Instead, it opened a temp file

File: src/main/java/de/nordgedanken/auto_hotkey/sdk/AhkSdkType.java (hunk 1 above)
Comment (takanuva15) [reply src/main/java/de/nordgedanken/auto_hotkey/sdk/AhkSdkType.java:63]: Yes, I was thinking the same thing. I should've added a TODO comment with that. I already wrote the script to do it; it's an easy one-liner that prints the version to the console:
`FileAppend %A_AhkVersion%, *`
Then we'd just need to execute it and read the console and save that into the SdkType when the user creates a new Sdk

**Comment by takanuva15** [comment]
> Is it also possible to instead of opening the warning to set the sdk open a dialog to set it? (Not sure if possible but would be nice if it is. Kind of thinking out loud).

I will look into this. There needs to be some kind of error alert shown to the user (otherwise the user has no idea why the Sdk dialog opened); using a message dialog is the easiest way to show it.

Eventually this error code will be deleted in a future release once an Sdk selector is added inside the run config itself (like what the Python plugin has).

**Comment by takanuva15** [comment]
> Is it possible to auto-detect the script path using the open file? I saw other languages do it but I am not sure how.

I am not sure what exactly you mean by this. Are you talking about running a file by clicking the green arrow next to the line numbers within the editor? (Could you give me an example of what steps are being taken?)

---

> I think this is introduced by your PR (I will double-check it). But it seems like the project doesn't have a file tree anymore if I use the open function in the welcome menu (not the import). Instead, it opened a temp file

I checked this against the master branch without my changes, and the same bug is happening. I also checked with a random python file, and the same issue is happening there too:
image
Thus I think this is an IntelliJ issue. In order to properly open a script, the user should click on the folder they want to open as a project, instead of the script file itself.

---


> Is it also possible to instead of opening the warning to set the sdk open a dialog to set it? (Not sure if possible but would be nice if it is. Kind of thinking out loud).

I have fixed this in my latest commit. If you click "OK" on the warning that shows, it will open up the project settings where you can setup/select the SDK.

As a random note, I also added a small test case to verify how tests work. (Still a lot of documentation to read for that)

**Pull request merged**